docs/reference/advanced_configuration.rst

```diff
@@ -444,18 +448,114 @@ some entries inside the  `$accessMapping` array in the linked Admin::
+        public function isGranted(AdminInterface $admin, $attributes, ?object $object = null): bool
+        {
+            return $this->customAccessLogic();
+        }
+
+        public function getBaseRole(AdminInterface $admin): string
+        {
+            return '';
+        }
+
+        public function buildSecurityInformation(AdminInterface $admin): array
+        {
+            return [];
+        }
+
+        public function createObjectSecurity(AdminInterface $admin, object $object): void
+        {
+        }
+
+        public function deleteObjectSecurity(AdminInterface $admin, object $object): void
         {
-            $this->customAccessLogic();
         }
     }
 
+Use `security_handler` tag to point to your custom SecurityHandler service
+for specific Admin class:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+
+        services:
+            # ...
+            admin.custom:
+                class: App\Admin\CustomAdmin
+                arguments: [~, App\Entity\Custom, ~]
+                tags:
+                    - { name: sonata.admin, manager_type: orm, label: Category, security_handler: App\Security\Handler\CustomSecurityHandler }
+
+You can also use the default SecurityHandler (defined in global configuration) 
+in your custom SecurityHandler::
+
+    // src/Security/Handler/CustomSecurityHandler.php
+
+    use Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface;
+
+    final class CustomSecurityHandler extends SecurityHandlerInterface
+    {
+        private SecurityHandlerInterface $defaultSecurityHandler;
+
+        public function __construct(SecurityHandlerInterface $defaultSecurityHandler)
+        {
+            $this->defaultSecurityHandler = $defaultSecurityHandler;
+        }
+
+        public function isGranted(AdminInterface $admin, $attributes, ?object $object = null): bool
+        {
+            // Custom access logic
+            if (...) {
+                return false;
+            }
+
+            // Default access logic
+            return $this->defaultSecurityHandler->isGranted($admin, $attributes, $object);
+        }
+
+        // ...
+    }
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+
+        services:
+            # ...
+            App\Security\Handler\CustomSecurityHandler:
+                arguments:
+                    - '@sonata.admin.security.handler'
+
+If you have a lot of SecurityHandler services that use the default SecurityHandler service,
+you can define a service alias:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+
+        services:
+            # ...
+            Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface: '@sonata.admin.security.handler'
+
+This way, you do not need to define each custom SecurityHandler service to specify 
+the default SecurityHandler service as an argument.
+
```

Comment:
Not sure we need this, this is classic symfony configuration and not needed with autowire/autoconfigure if I'm right.

Reply:
When I tested it and didn't register a custom SecurityHandler to inject the default ServiceHandler service explicit, I was getting an error like:
![qqqqqq](https://user-images.githubusercontent.com/16608636/140039804-36c658d0-5a40-4572-844d-363a598884f5.png)

I also checked what classes are available via autowiring and I did not find the SecurityHandlerInterface there:
![wewewewe](https://user-images.githubusercontent.com/16608636/140040886-26d03775-3911-486f-bcff-74819c50ea78.png)

Reply:
I see, thanks.